File: src/main/java/io/github/chindeaone/collectiontracker/config/categories/Mining.java
package io.github.chindeaone.collectiontracker.config.categories;

import com.google.gson.annotations.Expose;
import io.github.chindeaone.collectiontracker.config.categories.coleweight.ColeweightConfig;
import io.github.chindeaone.collectiontracker.config.categories.mining.*;
import io.github.chindeaone.collectiontracker.config.categories.mining.routes.MiningRoutesConfig;
import io.github.chindeaone.collectiontracker.config.categories.overlay.CommissionsConfig;
import io.github.chindeaone.collectiontracker.config.categories.overlay.MiningStatsConfig;
import io.github.notenoughupdates.moulconfig.annotations.Accordion;
import io.github.notenoughupdates.moulconfig.annotations.Category;
import io.github.notenoughupdates.moulconfig.annotations.ConfigOption;

public class Mining {

    @Expose
    @ConfigOption(name = "Coleweight", desc = "")
    @Accordion
    public ColeweightConfig coleweightConfig = new ColeweightConfig();

    @Expose
    @ConfigOption(name = "Commissions", desc = "")
    @Accordion
    public CommissionsConfig commissionsConfig = new CommissionsConfig();

    @Expose
    @ConfigOption(name = "Mining Stats Overlay", desc = "")
    @Accordion
    public MiningStatsConfig miningStatsConfig = new MiningStatsConfig();

    @Expose
    @ConfigOption(name = "HOTM perks", desc = "")
    @Accordion
    public HotmConfig hotmConfig = new HotmConfig();

    @Expose
    @Category(name = "Mining Routes", desc = "")
    public MiningRoutesConfig miningRoutesConfig = new MiningRoutesConfig();

    @Expose
    @ConfigOption(name = "Lantern Deployable", desc = "")
    @Accordion
    public LanternDeployableConfig lanternDeployableConfig = new LanternDeployableConfig();

    @Expose
    @ConfigOption(name = "Temporary Buffs Tracker", desc = "")
    @Accordion
    public TemporaryBuffsConfig temporaryBuffsConfig = new TemporaryBuffsConfig();
}

File: src/main/java/io/github/chindeaone/collectiontracker/config/categories/Mining.kt
package io.github.chindeaone.collectiontracker.config.categories

import com.google.gson.annotations.Expose
import io.github.chindeaone.collectiontracker.config.categories.coleweight.ColeweightConfig
import io.github.chindeaone.collectiontracker.config.categories.mining.HotmConfig
import io.github.chindeaone.collectiontracker.config.categories.mining.LanternDeployableConfig
import io.github.chindeaone.collectiontracker.config.categories.mining.TemporaryBuffsConfig
import io.github.chindeaone.collectiontracker.config.categories.mining.routes.MiningRoutesConfig
import io.github.chindeaone.collectiontracker.config.categories.overlay.CommissionsConfig
import io.github.chindeaone.collectiontracker.config.categories.overlay.MiningStatsConfig
import io.github.notenoughupdates.moulconfig.annotations.Accordion
import io.github.notenoughupdates.moulconfig.annotations.Category
import io.github.notenoughupdates.moulconfig.annotations.ConfigOption

class Mining {
    @Expose
    @ConfigOption(name = "Coleweight", desc = "")
    @Accordion
    var coleweightConfig: ColeweightConfig = ColeweightConfig()

    @Expose
    @ConfigOption(name = "Commissions", desc = "")
    @Accordion
    var commissionsConfig: CommissionsConfig = CommissionsConfig()

    @Expose
    @ConfigOption(name = "Mining Stats Overlay", desc = "")
    @Accordion
    var miningStatsConfig: MiningStatsConfig = MiningStatsConfig()

    @Expose
    @ConfigOption(name = "HotM Perks", desc = "")
    @Accordion
    var hotmConfig: HotmConfig = HotmConfig()

    @Expose
    @Category(name = "Mining Routes", desc = "")
    var miningRoutesConfig: MiningRoutesConfig = MiningRoutesConfig()

    @Expose
    @ConfigOption(name = "Lantern Deployable", desc = "")
    @Accordion
    var lanternDeployableConfig: LanternDeployableConfig = LanternDeployableConfig()

    @Expose
    @ConfigOption(name = "Temporary Buffs Tracker", desc = "")
    @Accordion
    var temporaryBuffsConfig: TemporaryBuffsConfig = TemporaryBuffsConfig()
}